### PR TITLE
Move PublishTransaction high fee check to caller

### PR DESCRIPTION
**Pull request by jrick**
We wish to add a JSON-RPC method which calls the wallet's
PublishTransaction method, but the enabling or disabling of high fee
checks works differently between the JSON-RPC and gRPC servers.  In
the JSON-RPC server, the method has its own default value for this
setting rather than using the wallet application config, while the
gRPC server uses the application settings.  One possible way to deal
with this would be to add another bool parameter for the setting to
the PublishTransaction signature.  This change instead keeps the
function signature the same, but requires callers to perform high fee
checks (if any), and gives more flexibility as to how this check is
enabled.

### CODE REVIEW

**Comment by jrick** [comment]
@itswisdomagain you will want to rebase #1846 over this after it is merged. After a bit of thought, I didn't want to add a bool parameter for the setting to the PublishTransaction method.

File: wallet/txrules/rules.go [hunk 1]
```diff
@@ -103,3 +103,21 @@ func PaysHighFees(totalInput dcrutil.Amount, tx *wire.MsgTx) bool {
 	maxFee := FeeForSerializeSize(1000*DefaultRelayFeePerKb, tx.SerializeSize())
 	return fee > maxFee
 }
+
+// TxPaysHighFees checks whether the signed transaction pays insanely high fees.
+// Transactons are defined to have a high fee if they have pay a fee rate that
+// is 1000 time higher than the default fee.  Total transaction input value is
```
Comment (itswisdomagain) [inline wallet/txrules/rules.go:109]:
```suggestion
// Transactons are defined to have a high fee if they pay a fee rate that
// is 1000 times higher than the default fee rate.  Total transaction input value is
```

File: wallet/txrules/rules.go [hunk 2]
```diff
@@ -103,3 +103,21 @@ func PaysHighFees(totalInput dcrutil.Amount, tx *wire.MsgTx) bool {
...
+// determined by summing the ValueIn fields of each input, and an error is returned
+// if any input values were the null value.
+func TxPaysHighFees(tx *wire.MsgTx) (bool, error) {
+	var input dcrutil.Amount
```
Comment (itswisdomagain) [inline wallet/txrules/rules.go:113]: totalInput?

File: wallet/txrules/rules.go [hunk 3]
```diff
@@ -103,3 +103,21 @@ func PaysHighFees(totalInput dcrutil.Amount, tx *wire.MsgTx) bool {
...
+	for i, in := range tx.TxIn {
+		if in.ValueIn < 0 {
```
Comment (itswisdomagain) [inline wallet/txrules/rules.go:115]: Can a valid tx input have 0 value?

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: wallet/txrules/rules.go (hunk 2 above)
Comment (jrick) [reply wallet/txrules/rules.go:113]: doesn't really matter for such a short function

File: wallet/txrules/rules.go (hunk 3 above)
Comment (jrick) [reply wallet/txrules/rules.go:115]: yes

**Review by itswisdomagain — APPROVED** [review]

**Pull request merged**